Ignore unexported fields

oscartbeaumont opened the pull request:
This change prevents fields that are unexported (start with a lowercase letter) from being processed by the package. This prevents a reflection error.

It does not cause any backwards compatibility issues with the existing codebase because with the current code you received an error if an unexported field exists on the arguments struct. This change prevents that error allowing the rest of the struct to be processed by skipping the unexported field.

A fix for issue #132

alexflint commented on parse.go:
```diff
@@ -263,7 +264,7 @@ func cmdFromStruct(name string, dest path, t reflect.Type) (*command, error) {
 		// Check for the ignore switch in the tag
 		tag := field.Tag.Get("arg")
-		if tag == "-" {
+		if tag == "-" || (unicode.IsLetter(rune(field.Name[0])) && unicode.IsLower(rune(field.Name[0]))) {
```
Would you add a check that `len(field.Name) > 0`?

And could this be factored this out into a function named `isExported`?

oscartbeaumont replied:
I have done as reccomended but with an `isUnexportedField` function.

alexflint reviewed (commented):
Thank you for this @oscartbeaumont!

Would you add a unit test for this? You could just duplicate one of the basic unittests and add an unexported field, then do a `require.NoPanic`

After that I'm ready to merge this.

oscartbeaumont commented:
I have added the unit test too. I don't know what you mean with `require.NoPanic` but the unit test works with the changes I have implemented and will only panic if they are removed which will alert you to a test failure so I believe its working as intended.

alexflint commented:
Hey @oscartbeaumont for some reason the diff is now showing up as if the entire file was modified. I don't see why this is happening (doesn't appear to be whitespace-related afaict) but I'd like to reduce the diff in order to preserve line-by-line git history. Can you figure out what's going on here and reduce the diff?

alexflint commented:
Closed in favor of #136

oscartbeaumont commented:
Thanks, sorry I have been super busy and never got to fixing the commit.

alexflint commented:
No problem Oscar!

On Mon, Feb 1, 2021 at 5:57 AM Oscar Beaumont <notifications@github.com>
wrote:

> Thanks, sorry I have been super busy and never got to fixing the commit.
>
> —
> You are receiving this because you modified the open/close state.
> Reply to this email directly, view it on GitHub
> <https://github.com/alexflint/go-arg/pull/133#issuecomment-770875201>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAE4J55Y6OUWN4OTA62ZA6LS42XMRANCNFSM4VJVFKLA>
> .
>
